### PR TITLE
fix off by one error using negative indices for diff'ing

### PR DESCRIPTION
**Pull request by changhiskhan**

### CODE REVIEW

File: python/lance/__init__.py [hunk 1]
```diff
@@ -107,11 +107,11 @@ def diff(dataset: FileSystemDataset, v1: int, v2: int = None) -> LanceDiff:
         dataset
     """
     if v1 < 0:
-        v1 = dataset.versions()[v1]["version"]
+        v1 = dataset.versions()[v1-1]["version"]
```
Comment (eddyxu) [inline python/lance/__init__.py:110]: Can you explain the logic here?  so `-1` is actually access `versions[-2]`?
means `latest == version[0]`?

Comment (changhiskhan) [reply python/lance/__init__.py:110]: yeah, it's a bit confusing. Version=0 being latest is the existing code convention. I wasn't thinking about the indexing as a list index but offsets from latest.

Comment (eddyxu) [reply python/lance/__init__.py:110]: Lets add them into document?

**Review by eddyxu — APPROVED** [review]
Pending docstring changes. The rest LGTM

**Pull request merged**